core/classes/Events/AbstractEvent.php

```diff
@@ -65,9 +55,9 @@ final public function params(): array
      * EventHandler::executeEvent(new EventClass(...));
      * ```.
      *
-     * @return array|null The response from the event, or null if the event is not returning a response
+     * @return AbstractEvent The response from the event
      */
-    final public static function dispatch(): ?array
+    final public static function dispatch(): AbstractEvent
```

Comment:
let's 🔥 this method, it was never used and defeats the purpose of the nice type checking we have now